dask/dataframe/tests/test_dataframe.py

```diff
@@ -3712,6 +3746,41 @@ def test_values():
     assert_eq(df.index.values, ddf.index.values)
 
 
+def test_values_extension_array():
+    from dask.array.utils import assert_eq
+
+    # NOTE: based on the list of pandas extension types at
+    # https://pandas.pydata.org/docs/user_guide/basics.html#dtypes
+    df = pd.DataFrame(
+        {
+            "datetime_tz": pd.array(
+                ["2020-01-01", "2020-02-01", "2020-03-01"],
+                dtype="datetime64[ns, US/Mountain]",
+            ),
+            "categorical": pd.Series(["a", "b", "a"], dtype="category"),
+            "period": pd.period_range("1/1/2021", "3/1/2021", freq="M"),
+            "sparse": pd.arrays.SparseArray([np.nan, np.nan, 1]),
+            "interval": pd.arrays.IntervalArray(pd.interval_range(start=0, end=3)),
+            "null_int": pd.array([1, 2, np.nan], dtype=pd.Int64Dtype()),
+            "string": pd.array(["foo", "bar", "baz"], dtype="string"),
+            "null_bool": pd.array([True, np.nan, False], dtype="boolean"),
+        }
+    )
+    ddf = dd.from_pandas(df, 2)
+
+    # HACK: compare the arrays as strings, since `assert_eq` and its NumPy relatives
+    # don't know how to handle pandas NA values. Because the dtype ends up being `object`,
+    # it's just doing elementwise comparison, and NA != NA, but also NA is not NaN.
+    assert str(df.values) == str(ddf.values.compute()), "Stringified values are not equal"
```

Comment:
What should I use instead of `assert_eq` for comparing NumPy `dtype=object` arrays containing pandas `NA` values? (Here and at the `FIXME` below.)

Reply:
Probably best to check the valid and NA values separately. Use numpy.testing for the valid ones, and `pd.isna(all)` for the NA ones.